### PR TITLE
Add date autosuggest to readme; Add en-gb support

### PR DESCRIPTION
**Pull request by liamcain**
solves #18
solves #30 

- README changes
- Chrono now looks at the moment.js locale to determine what it should use (only en-us and en-gb supported currently)
- Minor improvements to date autosuggest
  - Added new setting `autosuggestLinkToggle` instead of relying on the value of `modalLinkToggle`. This seemed more intuitive to me, but let me know what you think @argenos
  - Only register a CodeMirror event if the autosuggest is enabled
  - properly unregister event when the plugin is disabled/unloaded
  - Fix regex for `X days ago` so it shows up in the autosuggest again
- I added `eslint` to catch all the Typescript warnings. For a smaller diff, don't look at the last commit.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,73 +1,85 @@
...
 Starting on v0.3.2, in addition to the hotkey to parse the selected date, the following commands are also available (note that hotkeys are unset by default starting on v0.4.1):
 
-#### Natural Language Dates: Date Picker
-Opens the date picker menu
-<br>
-<img src="https://user-images.githubusercontent.com/5426039/99131292-0d0b2380-2613-11eb-8469-20d510fa2074.png" alt="Three commands to insert the current date, current time and a combination of both." style="zoom:75%;" />
-
-#### Natural Language Dates: Insert current date
-Inserts the current date, using the format specified in the settings menu. (default output `YYYY-MM-DD`)
+### Natural Language Dates: Date Picker
```
Comment (argenos) [inline README.md:42]:
```suggestion
#### Natural Language Dates: Date Picker

```

File: README.md [hunk 2]
```diff
@@ -1,73 +1,85 @@
...
-| `day`                          | natural language date string |
+| `obsidian://nldates` Parameter | Description                             |
+| ------------------------------ | --------------------------------------- |
+| `day`                          | natural language date string            |
 | `newPane`                      | open note in new pane, default is `yes` |
 
-## Commands and hotkeys
+### Commands and Hotkeys
 
 Starting on v0.3.2, in addition to the hotkey to parse the selected date, the following commands are also available (note that hotkeys are unset by default starting on v0.4.1):
```
Comment (argenos) [inline README.md:40]:
```suggestion
`nldates` adds a few commands to work with dates in natural language. You can add custom hotkeys for them by going to `Settings > Hotkeys` and filtering by `Natural Language Dates` (Note that hotkeys are unset by default starting on **v0.4.1**).

```

File: src/main.ts [hunk 3]
```diff
@@ -174,46 +194,46 @@ export default class NaturalLanguageDates extends Plugin {
...
 
-  onTrigger(mode: string) {
-    let activeLeaf: any = this.app.workspace.activeLeaf;
-    let editor = activeLeaf.view.sourceMode.cmEditor;
-    var cursor = editor.getCursor();
-    var selectedText = this.getSelectedText(editor);
-
-    let date = this.parseDate(selectedText);
+  onTrigger(mode: string): void {
+    const activeView = this.app.workspace.getActiveViewOfType(MarkdownView);
```
Comment (argenos) [inline src/main.ts:229]: I could have sworn I had fixed this already! Thanks ❤️

**Review by argenos — COMMENTED** [review]
Looks good to me! I only added two comments with minor suggestions.

**Comment by argenos** [comment]
I agree! I think it makes sense to separate these two contexts. Ready to merge?

**Comment by liamcain** [comment]
Yep, feel free to merge!

**Pull request merged**